Attestation/ek-secret.md

```diff
@@ -0,0 +1,173 @@
+# `Passing a secret to a TPM using only the EK`
+
+This is example code to pass a secret to a system by just knowing the endorsenment key.
+We will be using the current (commit 07a92e9fa75548ea102ce90b3b6182093b3f7a73 or later) master branch of https://github.com/tpm2-software/tpm2-pytss
+
+The terms for the systems are `remote system`, the system we want to pass the secret to and `local system`, the system which has the secret but doesn't have any TPM.
+One assumtion that will be made is that you already have the EK for the remote system on the local system, and trust it.
+
+## Concept
+The concept is that the key being activated by a call to ActivateCredential on the remote system doesn't have to be generated on the TPM, as long as the public and private parts are loaded it will succeed.
+By generating a temporary key pair on the local system we can run MakeCredential with the remote system EK, name of the locally generated key and the secret.
+
+## Local system script
+```
+#!/usr/bin/python3
+
+import sys
+from tpm2_pytss import *
+from tpm2_pytss.makecred import MakeCredential
+from cryptography.hazmat.primitives.asymmetric import ec
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.serialization import Encoding, PublicFormat, PrivateFormat, NoEncryption
+
+def main(ekpath, publicpath, sensitivepath, credpath, secretpath, oursecret):
+    # first read the EK and unmarshal it
+    with open(ekpath, 'rb') as ef:
+        ekb = ef.read()
+    ekpub, _ = TPM2B_PUBLIC.Unmarshal(ekb)
+
+    # Now we generate the temporary key pair
```

Comment:
See above comment about separating this so that we can keep reusing a single well-known key.